components/automate-ui/src/app/pages/project/details/project-details.component.spec.ts

```diff
@@ -0,0 +1,199 @@
+        MockComponent({ selector: 'chef-breadcrumb', inputs: ['link'] }),
+        MockComponent({ selector: 'chef-tab-selector', inputs: ['value'] }),
+        MockComponent({ selector: 'chef-button', inputs: ['disabled'] }),
+        MockComponent({ selector: 'chef-heading' }),
+        MockComponent({ selector: 'chef-loading-spinner' }),
+        MockComponent({ selector: 'chef-option' }),
+        MockComponent({ selector: 'chef-page-header' }),
+        MockComponent({ selector: 'chef-subheading' }),
+        MockComponent({ selector: 'chef-table' }),
+        MockComponent({ selector: 'chef-thead' }),
+        MockComponent({ selector: 'chef-tbody' }),
+        MockComponent({ selector: 'chef-tr' }),
+        MockComponent({ selector: 'chef-th' }),
+        MockComponent({ selector: 'chef-td' }),
+        MockComponent({ selector: 'ng-container', inputs: ['hidden'] }),
+        ProjectDetailsComponent
+      ],
+      imports: [
+        ReactiveFormsModule,
+        RouterTestingModule,
+        ChefPipesModule,
+        StoreModule.forRoot({
+          router: () => ({
+            state: {
+              url: '/projects/uuid-1',
+              queryParams: {},
+              params: {},
+              fragment: '',
+              path: ['/']
+            },
+            previousRoute: {},
+            navigationId: 0
+          }),
+          projects: projectEntityReducer
+        })
+      ],
+      providers: [
+        FeatureFlagsService
+      ]
+    }).compileComponents();
+  }));
+
+  let store: Store<NgrxStateAtom>;
+  beforeEach(() => {
+    store = TestBed.get(Store);
+
+    store.dispatch(new GetProjectSuccess({
+      project: {
+        id: 'uuid-1', name: 'Default',
+        type: <IAMType>'CHEF_MANAGED'
+      }
+    }));
+
+    jasmine.addMatchers(customMatchers);
+    fixture = TestBed.createComponent(ProjectDetailsComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  describe('when there are no rules', () => {
+    beforeEach(() => {
+      store.dispatch(new GetRulesSuccess({
+          rules: []
+      }));
+      component.rules = [];
+      fixture.detectChanges();
+    });
+
+    it('rules array should be empty', () => {
+      expect(component.rules.length).toBe(0);
+    });
+
+    it('show rules section when rules tab is selected', () => {
+      component.onTabChange({ target: { value: 'rules'} });
+      expect(component.showTab('rules')).toBeTruthy();
+    });
+
+    it('should not display rule table when no rules', () => {
+      expect(component.showRulesTable()).toBeFalsy();
+    });
+
+    it('the create your first rule message should display', () => {
+      expect(component.showFirstRuleMessage()).toBeTruthy();
+    });
+  });
+
+  describe('when there are rules', () => {
+    beforeEach(() => {
+      store.dispatch(new GetRulesSuccess({
+          rules: rules
+      }));
+      component.project = project;
+      component.rules = rules;
```

Comment:
The `dispatch` here is not doing anything--you can see that by commenting it out and you'll see that all tests still pass. That means you are not actually testing the block beginning `this.store.select(projectFromRoute)...` in the component. That would be worthwhile but would require some rework on that block if you wanted to do it. 
For the tests you have here, then, I would suggest delete the `dispatch` above and also delete the first test `rules array should have two rules` because it is really a tautology. The remaining tests here are useful with just setting the properties on L169-170 above.
(These comments applies to the prior section on "no rules", too.)